modules/services/ihaskell.nix

```diff
@@ -0,0 +1,113 @@
+# Adapted from Nixpkgs.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.ihaskell;
+  ihaskellSrc = pkgs.fetchFromGitHub {
+    owner = "gibiansky";
+    repo = "IHaskell";
+    rev = "c2cb8e6789c3f2485ba9e1e0436177b700edb227";
+    sha256 = "0h5hpdfa3daabf1ijl7y5qnpy60yrf7h0ws2n3v5przfmgr8phv5";
+  };
+  ihaskellResult = (import (ihaskellSrc + /release.nix) {
+    compiler = cfg.compiler;
+    packages = cfg.extraPackages;
+  });
+
+in
+
+{
+  meta.maintainers = [ maintainers.rycee ];
+
+  options.services.ihaskell = {
+    enable = mkOption {
+      default = false;
+      description = "Autostart an IHaskell notebook service.";
+    };
+
+    compiler = mkOption {
+      default = "ghc864";
+      example = literalExample ''
+        ghc864
+      '';
+      description = ''
+        The Haskell compiler to use
+      '';
+    };
+
+    notebooksPath = mkOption {
+      default = "$HOME/ihaskell";
+      example = literalExample ''
+        $HOME/projects/ihaskell-notebooks
+      '';
+      description = ''
+        Directory where iHaskell will store notebooks
```

Comment:
The project name should be used consistently, in [their readme](https://github.com/gibiansky/IHaskell/blob/master/README.md) they seem to prefer IHaskell. So I would suggest to change iHaskell and ihaskell to IHaskell throughout the module.